src/itemLogin/ItemLoginScreen.tsx

```diff
@@ -107,9 +107,9 @@ const ItemLoginScreen: FC<ItemLoginScreenProps> = ({
 }) => {
   const { t } = useTranslation();
   const loginModeRef = useRef(null);
-  const [password, setPassword] = useState<string | undefined>(undefined);
-  const [username, setUsername] = useState<string | undefined>(undefined);
-  const [memberId, setMemberId] = useState<string | undefined>(undefined);
+  const [password, setPassword] = useState<string>();
+  const [username, setUsername] = useState<string>();
+  const [memberId, setMemberId] = useState<string>();
```

Comment:
Doesn't `useState` hold `undefined` by default if no argument is passed?
In which case the type is thus technically wrong: it should either be `useState<string | undefined>();` or `useState<string>("");` I think.

Reply:
You are right that `useState` holds undefined as the default value. But they typed it so that you only pass the other type as the generic argument as can be seen below:
```ts
    /**
     * Returns a stateful value, and a function to update it.
     *
     * @version 16.8.0
     * @see https://reactjs.org/docs/hooks-reference.html#usestate
     */
    function useState<S>(initialState: S | (() => S)): [S, Dispatch<SetStateAction<S>>];
    // convenience overload when first argument is omitted
    /**
     * Returns a stateful value, and a function to update it.
     *
     * @version 16.8.0
     * @see https://reactjs.org/docs/hooks-reference.html#usestate
     */
    function useState<S = undefined>(): [S | undefined, Dispatch<SetStateAction<S | undefined>>];
```
So with `useState<string>()` you get a variable that is `string | undefined` and if you specify the value of the argument then it will be just the type of the argument.

Very good comment though, TIL how `useState` is typed !

Reply:
You are right, I somehow got confused and thought this was equivalent to `const [state, setState]: string = useState<string>()` but this would obviously be `const [state, setState]: string | undefined = useState<string>()`, and anyway we let TS infer. My bad! 😕